src/core/Akka.Remote/Transport/Streaming/NetworkStreamTransport.cs

```diff
@@ -0,0 +1,289 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Configuration;
+
+namespace Akka.Remote.Transport.Streaming
+{
+    public class NetworkStreamTransportSettings : StreamTransportSettings
+    {
+        public string Hostname { get; }
+
+        public int Port { get; }
+
+        public IPAddress BindIp { get; }
+
+        public int BindPort { get; }
+
+        public TimeSpan ConnectionTimeout { get; }
+
+        public int SendBufferSize { get; }
+
+        public int ReceiveBufferSize { get; }
+
+        public int Backlog { get; }
+
+        public bool TcpNoDelay { get; }
+
+        public bool TcpKeepAlive { get; }
+
+        public NetworkStreamTransportSettings(Config config)
+            : base(config)
+        {
+            Hostname = config.GetString("hostname");
+
+            if (string.IsNullOrEmpty(Hostname))
+                Hostname = "localhost";
+
+            Port = config.GetInt("port");
+
+            Hostname = config.GetString("hostname");
+            Port = config.GetInt("port");
+
+            string bindHostname = config.GetString("bind-hostname");
+
+            if (string.IsNullOrEmpty(bindHostname))
+                bindHostname = Hostname;
+
+            IPAddress ip;
+            if (Hostname.Equals("localhost", StringComparison.OrdinalIgnoreCase))
+            {
+                BindIp = IPAddress.Loopback;
+            }
+            else if (IPAddress.TryParse(bindHostname, out ip))
+            {
+                // The socket is using DualMode, if ipv6 is supported, listening on IPv6Any
+                // will work for any address (ipv4 and ipv6)
+                if (IPAddress.Any.Equals(ip) && Socket.OSSupportsIPv6)
+                    ip = IPAddress.IPv6Any;
+
+                BindIp = ip;
+            }
+            else
+            {
+                // If hostname is not localhost or an ip, bind to IPAny
+                BindIp = Socket.OSSupportsIPv6 ? IPAddress.IPv6Any : IPAddress.Any;
```

Comment:
Actually - nevermind. 0.0.0.0 is a fine default - consistent with what JVM Akka and Helios currently do.